File: npm/src/lib/prompt.ts
import * as readline from 'readline/promises';

export async function ask(question: string): Promise<string> {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });
  try {
    const answer = await rl.question(question);
    return answer.trim();
  } finally {
    rl.close();
  }
}

export async function confirm(question: string): Promise<boolean> {
  const answer = await ask(`${question} (y/N) `);
  return answer.toLowerCase() === 'y' || answer.toLowerCase() === 'yes';
}

File: python/src/formhandle/prompt.py
"""Interactive prompt helpers."""


def ask(question: str) -> str:
    try:
        return input(question).strip()
    except (EOFError, KeyboardInterrupt):
        print()
        raise SystemExit(1)


def confirm(question: str) -> bool:
    answer = ask(f"{question} (y/N) ")
    return answer.lower() in ("y", "yes")
